src/app/search/query.js

```diff
@@ -29,19 +29,25 @@ const parseQuery = (queryString) => {
  * @param {string} query, {string} field
  * @return {object}
  */
-export const buildQueryBody = (query, field = 'keyword') => {
-  if (!query) {
+export const buildQueryBody = (query) => {
+  if (!query.query) {
     throw new Error('A valid query string must be passed');
   }
 
-  let queryField = field;
-  if (_isArray(field)) {
-    queryField = field.join('|');
+  let queryField = query.field || initialSearchQuery.field;
+  if (_isArray(query.field)) {
+    queryField = query.field.join('|');
   }
-  const parsedQuery = parseQuery(query);
-  return { query: parsedQuery, field: queryField };
+  const parsedQuery = parseQuery(query.query);
+  return Object.assign({}, initialSearchQuery, query, { query: parsedQuery, field: queryField });
 };
 
+// export const getQueryString = query => new URLSearchParams(query).toString(); // not supported
+
+export const getQueryString = query => Object.keys(query)
+  .map(key => [key, query[key]].map(encodeURIComponent).join('='))
```

Comment:
Encoding, yes!! :)

Reply:
URLSearchParams is more solid as it's browser implemented, but our version of node doesn't supports it yet.